tools/test_api.py

```diff
@@ -2159,7 +2161,9 @@ def build_tests(tests, base_source_paths, build_path, target, toolchain_name,
             'verbose': verbose,
             'app_config': app_config,
             'build_profile': build_profile,
-            'silent': True
+            'silent': True,
+            # Coverage requires main(). So if enabled on any module, enable it on test as well.
+            'coverage_filter': ".*" if coverage_filter else []
```

Comment:
Yes it will enable coverage on whole test source. Since it would complicated to find the file containing ```main()```.